composer/trainer/trainer.py

```diff
@@ -279,27 +267,29 @@ def __init__(
         self.engine.run_event(Event.INIT)
 
         # Need to use hparams here because optimizer and schedulers need to be created after Event.INIT
-        if not optimizer_hparams:
+        if len(ensure_tuple(optimizer_hparams)) == 0:
             optimizer_hparams = DecoupledSGDWHparams(lr=0.1, momentum=0.9, weight_decay=1.0e-4)
-        if not schedulers_hparams:
+        if len(ensure_tuple(schedulers_hparams)) == 0:
             schedulers_hparams = [CosineAnnealingLRHparams(T_max=f"{max_epochs}ep"), WarmUpLRHparams()]
-        if not isinstance(schedulers_hparams, list):
-            schedulers_hparams = [schedulers_hparams]
-        optimizer = optimizer_hparams.initialize_object(param_group=self.state.model.parameters())
+        optimizers = [
+            optimizer_hparams.initialize_object(param_group=self.state.model.parameters())
+            for optimizer_hparams in ensure_tuple(optimizer_hparams)
+        ]
+        if len(optimizers) != 1:
+            raise NotImplementedError("Multiple optimizers are not supported.")
         schedulers = [
-            x.initialize_object(optimizer, self.state.steps_per_epoch) for x in ensure_warmup_last(schedulers_hparams)
+            x.initialize_object(optimizers[0], self.state.steps_per_epoch)
+            for x in ensure_warmup_last(list(ensure_tuple(schedulers_hparams)))
         ]
-        self.state.optimizers = optimizer
-        self.state.schedulers = ComposedScheduler(schedulers=schedulers)
+        self.state.optimizers = optimizers
+        self.state.schedulers = [ComposedScheduler(schedulers=schedulers)]
```

Comment:
Not sure, but the Composed Scheduler casting maybe could be the underlying schedulers object?  A little opinionated, but can auto set ComposedScheduler in the setter